R/functions-xcmsRaw.R

```diff
@@ -299,7 +299,7 @@ profEIC <- function(object, mzrange, rtrange = NULL, step = 0.1) {
                                                  yes = 0, no = NA),
                               shiftByHalfBinSize = TRUE,
                               sortedX = TRUE)
-            if (length(toIdx) == 1)
+            if (length(binRes) == 1)
```

Comment:
Couldn't it be that `binRes` is a `list`, just of length 1? Then this call would create a list of a list. Would:
```r
if (!is.list(binRes))
    binRes <- list(binRes)
```